RESTOpenSubtitlesHandler/RequestHandler.cs

```diff
@@ -1,100 +1,100 @@
+        private static int _requestCount;
+
+        public static void SetApiKey(string key)
+        {
+            if (_apiKey == string.Empty)
+            {
+                _apiKey = key;
+            }
+        }
+
+        public static async Task<(string response, (int remaining, int reset) limits, Dictionary<string, string> headers, HttpStatusCode statusCode)> SendRequestAsync(string endpoint, HttpMethod method, string body, Dictionary<string, string> headers, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(_apiKey))
+            {
+                throw new Exception("API key has not been set up");
+            }
+
+            headers ??= new Dictionary<string, string>();
+
+            if (!headers.ContainsKey("Api-Key"))
+            {
+                headers.Add("Api-Key", _apiKey);
+            }
+
+            var url = endpoint.StartsWith("/") ? BaseApiUrl + endpoint : endpoint;
+            var api = url.StartsWith(BaseApiUrl);
+
+            if (api)
+            {
+                if (_hRemaining == 0)
+                {
+                    await Task.Delay(1000 * _hReset, cancellationToken).ConfigureAwait(false);
+                    _hRemaining = -1;
+                    _hReset = -1;
+                }
+
+                if (_requestCount == 40)
+                {
+                    var diff = DateTime.UtcNow.Subtract(_windowStart).TotalSeconds;
+                    if (diff <= 10)
+                    {
+                        await Task.Delay(1000 * (int)Math.Ceiling(10 - diff), cancellationToken).ConfigureAwait(false);
+                    }
+                }
+
+                if (DateTime.UtcNow.Subtract(_windowStart).TotalSeconds >= 10)
+                {
+                    _windowStart = DateTime.UtcNow;
+                    _requestCount = 0;
+                }
+            }
+
+            var (response, responseHeaders, httpStatusCode) = await Util.SendRequestAsync(url, method, body, headers, cancellationToken).ConfigureAwait(false);
+
+            if (!api)
+            {
+                return (response, (_hRemaining, _hReset), responseHeaders, httpStatusCode);
+            }
+
+            _requestCount++;
+
+            if (responseHeaders.TryGetValue("x-ratelimit-remaining-second", out var value))
+            {
+                int.TryParse(value, out _hRemaining);
+            }
+
+            if (responseHeaders.TryGetValue("ratelimit-reset", out value))
+            {
+                int.TryParse(value, out _hReset);
+            }
+
+            if (httpStatusCode != HttpStatusCode.TooManyRequests)
+            {
+                return (response, (_hRemaining, _hReset), responseHeaders, httpStatusCode);
+            }
+
+            await Task.Delay(1000 * (_hReset == -1 ? 5 : _hReset), cancellationToken).ConfigureAwait(false);
+
+            return await SendRequestAsync(endpoint, method, body, headers, cancellationToken).ConfigureAwait(false);
+
```

Comment:
unnecessary line

Reply:
This if statement is reversed (look at the old code) I didn't change this still want me to remove it?

Reply:
yeah (the empty line below return statement)